### PR TITLE
Added UseCustomJsonConverters method to use custom json converters 

### LINKED ISSUE #17: RegisterType is broken
See https://github.com/mbdavid/LiteDB/wiki/Object-Mapping
```fsharp
    mapper.RegisterType<AutoSetHangtagTemplate>(
        serialize =
            fun value ->
                Bson.serializeField(value)
            ,
        deserialize =
            fun bsonValue ->
                let v = Bson.deserializeField bsonValue
                {v with RetrieveFrom = RetrieveFrom.FromDB}
    )
```
I cannot catch the code in `function` any more
Should we fix it?

### PR DESCRIPTION
**Pull request by humhei**
Fixes #17

### CODE REVIEW

File: LiteDB.FSharp/Json.fs [hunk 1]
```diff
@@ -80,6 +80,8 @@ type MapSerializer<'k,'v when 'k : comparison>() =
 
 module private Cache =
 
+    let customSerializer = Dictionary<Type, obj -> BsonDocument>()
```
Comment (humhei) [inline LiteDB.FSharp/Json.fs]: (obj -> BsonDocument) but not (obj -> BsonValue)

Comment (Zaid-Ajaj) [reply LiteDB.FSharp/Json.fs]: Maybe because `BsonDocument` is a `BsonValue`? I am not sure though

**Comment by Zaid-Ajaj** [comment]
Again, feel free to merge or let me know so I can merge/publish it

**Comment by humhei** [comment]
It seems that LiteDB (de)serialize directly with bson data and entity type(Also for properties)
While we going through a json data middleware by jsonconverters

Maybe i only want to be able to use custom json converter
So i can do more custom convertions

**Comment by Zaid-Ajaj** [comment]
@humhei I honestly wouldn't know why someone would want to customize the converter, we use JSON just as an internal layer to make the conversion easy to/from bson. In anycase, I will publish the new version  asap

**Pull request merged**